### PR TITLE
Fix crash when hitting 'T' to tap timing while no timing point is selected

### PR DESCRIPTION
**Pull request by peppy**
Addresses #21385.

### CODE REVIEW

File: osu.Game/Screens/Edit/Timing/TapButton.cs [hunk 1]
```diff
@@ -295,6 +295,9 @@ public void OnReleased(KeyBindingReleaseEvent<GlobalAction> e)
 
         private void handleTap()
         {
+            if (selectedGroup?.Value == null)
+                return;
```
Comment (bdach) [inline osu.Game/Screens/Edit/Timing/TapButton.cs:299]: This one looks redundant, as the only other usage of `selectedGroup` in this method is null-propagated and checked anyway.

https://github.com/ppy/osu/blob/8d0fe61aff00014dae8ef4fe1b0ad3fb0d896abf/osu.Game/Screens/Edit/Timing/TapButton.cs#L316-L322

File: osu.Game/Screens/Edit/Timing/TapTimingControl.cs [hunk 2]
```diff
@@ -183,12 +183,18 @@ private void load(OverlayColourProvider colourProvider, OsuColour colours)
 
         private void start()
         {
+            if (selectedGroup.Value == null)
+                return;
```
Comment (bdach) [inline osu.Game/Screens/Edit/Timing/TapTimingControl.cs:187]: Thoughts on adding these to `adjust{Bpm,Offset}` as well? Those cases don't look reachable from user interaction yet, but it's a bit weird to have null checked in only some branches.

**Comment by bdach** [comment]
I have also pushed basic test coverage for this. If https://github.com/ppy/osu/pull/21386#discussion_r1030562013 is acted upon, the added test can be extended to also cover the offset/BPM controls.

File: osu.Game/Screens/Edit/Timing/TapButton.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Screens/Edit/Timing/TapButton.cs:299]: I saw that, but it's only checking the DI level nullability (I guess for tests?), not the `Value` portion so saw value in adding this check.

File: osu.Game/Screens/Edit/Timing/TapTimingControl.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Screens/Edit/Timing/TapTimingControl.cs:187]: I wouldn't be against it.

Comment (bdach) [reply osu.Game/Screens/Edit/Timing/TapTimingControl.cs:187]: I'll go ahead and add them to the rest, then (with appropriate test coverage).

**Review by bdach — APPROVED** [review]

**Pull request merged**